source/extensions/filters/network/redis_proxy/conn_pool_impl.cc

```diff
@@ -122,6 +142,85 @@ InstanceImpl::ThreadLocalPool::makeRequest(const std::string& key,
     client->redis_client_->addConnectionCallbacks(*client);
   }
 
+  // cache host for quick lookup via address for redirection support
+  if (host->address()) {
+    auto hostCachedByAddress = host_address_map_.find(host->address()->asString());
+    if (hostCachedByAddress == host_address_map_.end()) {
+      host_address_map_[host->address()->asString()] = host;
+    }
+  }
+
+  return client->redis_client_->makeRequest(request, callbacks);
+}
+
+Common::Redis::Client::PoolRequest*
+InstanceImpl::ThreadLocalPool::makeRequestToHost(const std::string& host_address,
+                                                 const Common::Redis::RespValue& request,
+                                                 Common::Redis::Client::PoolCallbacks& callbacks) {
+  if (cluster_ == nullptr) {
+    ASSERT(client_map_.empty());
+    ASSERT(host_set_member_update_cb_handle_ == nullptr);
+    return nullptr;
+  }
+
+  auto colon_pos = host_address.rfind(":");
+  if ((colon_pos == std::string::npos) || (colon_pos == (host_address.size() - 1))) {
+    return nullptr;
+  }
+
+  std::string ip_address = host_address.substr(0, colon_pos);
+  bool ipv6 = (ip_address.find(":") != std::string::npos);
+  std::string host_address_map_key;
+  Network::Address::InstanceConstSharedPtr address_ptr;
+
+  if (!ipv6) {
+    host_address_map_key = host_address;
+  } else {
+    // find our canonical IPv6 address string representation by
+    // creating the address from the Redis server info...
+    std::string ip_port = host_address.substr(colon_pos + 1);
+    try {
+      address_ptr =
+          std::make_shared<Network::Address::Ipv6Instance>(ip_address, std::stoi(ip_port));
+      host_address_map_key = address_ptr->asString();
+    } catch (...) {
+      // bad address
+      return nullptr;
+    }
+  }
+
+  auto it = host_address_map_.find(host_address_map_key);
+  if (it == host_address_map_.end()) {
+    // not known to the cluster manager, create a new host and insert it into the map
+    try {
+      if (!ipv6) {
+        // only create an IPv4 address instance if we need a new Upstream::HostImpl
+        std::string ip_port = host_address.substr(colon_pos + 1);
+        address_ptr =
+            std::make_shared<Network::Address::Ipv4Instance>(ip_address, std::stoi(ip_port));
+      }
+      Upstream::HostSharedPtr new_host{new Upstream::HostImpl(
+          cluster_->info(), "", address_ptr, envoy::api::v2::core::Metadata::default_instance(), 1,
+          envoy::api::v2::core::Locality(),
+          envoy::api::v2::endpoint::Endpoint::HealthCheckConfig::default_instance(), 0,
+          envoy::api::v2::core::HealthStatus::UNKNOWN)};
+      host_address_map_[host_address_map_key] = new_host;
+    } catch (...) {
+      // can't resolve address, or some other problem
```

Comment:
Put this try/catch only around the address creation. Nothing else should fail.

Reply:
ok, will still catch more than EnvoyException, because of possible std::stoi() exceptions

Reply:
See above.

Reply:
ok